sdk/python/kfp/compiler/compiler.py

```diff
@@ -828,8 +830,18 @@ def _create_workflow(self,
     from ._data_passing_rewriter import fix_big_data_passing
     workflow = fix_big_data_passing(workflow)
 
-    import json
-    workflow.setdefault('metadata', {}).setdefault('annotations', {})['pipelines.kubeflow.org/pipeline_spec'] = json.dumps(pipeline_meta.to_dict(), sort_keys=True)
+    metadata = workflow.setdefault('metadata', {})
+    annotations = metadata.setdefault('annotations', {})
+    annotations['pipelines.kubeflow.org/sdk_version'] = 'kfp==' + kfp.__version__
+
+    # Adding kfp_sdk_version since sdk_version may be ambiguous when multiple SDKs are used at the same time.
+    annotations['pipelines.kubeflow.org/kfp_sdk_version'] = kfp.__version__
+    annotations['pipelines.kubeflow.org/pipeline_compilation_time'] = datetime.datetime.now().isoformat()
+    annotations['pipelines.kubeflow.org/pipeline_spec'] = json.dumps(pipeline_meta.to_dict(), sort_keys=True)
```

Comment:
Curious what's the purpose of pipeline_spec?

Reply:
Note that it was there before this PR which just moves it to a new line.

The purpose is to provide some needed information to the UX:
* Human-readable pipeline name and description.
* Pipeline inputs and their types

The UX can use the type information for validating the user input (not implemented yet). This was the main driver for including this information.

Reply:
Got it. Thanks